pyteal/ir/tealcomponent.py

```diff
@@ -22,6 +31,17 @@ def getSubroutines(self) -> List["SubroutineDefinition"]:
     def resolveSubroutine(self, subroutine: "SubroutineDefinition", label: str) -> None:
         pass
 
+    def stack_frames(self) -> NatalStackFrame:
+        from pyteal.ast import Expr
+
+        root_expr = self._sframes_container or self.expr
+        if root_expr:
+            if subroot := getattr(root_expr, "_sframes_container", None):
+                root_expr = cast(Expr, subroot)
+            return root_expr.stack_frames
+
+        return cast(NatalStackFrame, self._stack_frames)
```

Comment:
When there is a `self._sframes_container` it means that the source mapper wants to use that information instead of what `self.expr` is providing us. The source mapper is not allowed to _replace_ `self.expr` because it is crucial to the semantics of the program. However, I this code represents an very careful approach that I originally undertook when I first started this work. I've actually moved on from this approach and have been overwriting the `stack_frames` member of `Expr`'s instead. So I have added a task item to #681 to investigate the feasibility of removing `self._sframes_container` completely and instead overwrite `self.expr.stack_frames` OR `self._stack_frames`. 

See the task "Investigate the feasibility..." in #681